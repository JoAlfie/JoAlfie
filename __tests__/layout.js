import React from "react"
import renderer from "react-test-renderer"
import Layout from "../src/components/layout"

import * as Gatsby from "gatsby"
const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

useStaticQuery.mockImplementation(() => ({
	site: {
		siteMetadata: {
			tite: `Site title`,
			description: `Site meta description`,
			menuLinks: [],
		},
	},
}))

describe("Layout", () => {
	it("renders correctly without set description", () => {
		const site = {
			siteMetadata: {
				title: `Site title`,
				description: `Site meta description`,
				menuLinks: [],
			},
		}
		const tree = renderer
			.create(
				<Layout site={site} metaTitle="Hello world">
					hello world
				</Layout>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})

describe("Layout", () => {
	it("renders correctly with set description", () => {
		const site = {
			siteMetadata: {
				title: `Site title`,
				description: `Site meta description`,
				menuLinks: [],
			},
		}
		const tree = renderer
			.create(
				<Layout
					site={site}
					metaTitle="Hello world"
					metaDescription="Set meta description"
				>
					hello world
				</Layout>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
