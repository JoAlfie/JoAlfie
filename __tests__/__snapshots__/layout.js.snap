// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly with set description 1`] = `
<div
  className="container"
>
  <header
    className="mainheader"
  >
    <a
      className="mainheader__mainlink"
      href="/"
    />
    0
  </header>
  <main>
    hello world
  </main>
</div>
`;

exports[`Layout renders correctly without set description 1`] = `
<div
  className="container"
>
  <header
    className="mainheader"
  >
    <a
      className="mainheader__mainlink"
      href="/"
    />
    0
  </header>
  <main>
    hello world
  </main>
</div>
`;
